internal/semerrgroup/sem.go

```diff
@@ -30,6 +40,24 @@ func (s *Group) Go(fn func() error) {
 }
 
 // Wait waits for the group to complete and return an error if any.
-func (s *Group) Wait() error {
+func (s *parallelGroup) Wait() error {
 	return s.g.Wait()
 }
+
+var _ Group = &serialGroup{}
+
+type serialGroup struct {
+	err error
+}
+
+// Go execs one function at a time.
+func (s *serialGroup) Go(fn func() error) {
+	if err := fn(); err != nil {
```

Comment:
maybe here we should check if `s.err != nil` first?

if so, ignore the subsequent runs? 

The idea is to make the behavior similar to errgroup which afaik works that way 🤔

Reply:
done